qa/rbd/krbd_whole_object_discard.t

```diff
@@ -0,0 +1,120 @@
+
+  $ get_block_name_prefix() {
+  >     rbd info --format=json $1 | python -c "import sys, json; print json.load(sys.stdin)['block_name_prefix']"
+  > }
+
+  $ rbd create --size 200M img
+  $ DEV=$(sudo rbd map img)
+  $ xfs_io -c 'pwrite -b 4M 0 200M' $DEV >/dev/null
+  $ sudo rbd unmap $DEV
+  $ rbd snap create img@snap
+  $ rbd snap protect img@snap
+
+  $ rbd clone img@snap cloneimg1
+  $ DEV=$(sudo rbd map cloneimg1)
+  $ hexdump $DEV
+  0000000 cdcd cdcd cdcd cdcd cdcd cdcd cdcd cdcd
+  *
+  c800000
+  $ blkdiscard -l 100M $DEV
+  $ hexdump $DEV
+  0000000 0000 0000 0000 0000 0000 0000 0000 0000
+  *
+  6400000 cdcd cdcd cdcd cdcd cdcd cdcd cdcd cdcd
+  *
+  c800000
+  $ sudo rbd unmap $DEV
+  $ DEV=$(sudo rbd map cloneimg1)
+  $ hexdump $DEV
+  0000000 0000 0000 0000 0000 0000 0000 0000 0000
+  *
+  6400000 cdcd cdcd cdcd cdcd cdcd cdcd cdcd cdcd
+  *
+  c800000
+  $ sudo rbd unmap $DEV
+
+  $ rbd clone --stripe-unit 512K --stripe-count 7 img@snap cloneimg2
```

Comment:
Nit: might be nice to have a quick comment about what you are attempting to test w/ these magic stripe counts

Reply:
Just varying the object set size and asserting on the number of objects created.  I'll add the math for each clone.